pkg/image/importer/credentials.go

```diff
@@ -169,7 +169,7 @@ func (s *SecretCredentialStore) init() credentialprovider.DockerKeyring {
 func basicCredentialsFromKeyring(keyring credentialprovider.DockerKeyring, target *url.URL) (string, string) {
 	// TODO: compare this logic to Docker authConfig in v2 configuration
 	var value string
-	if len(target.Scheme) == 0 || target.Scheme == "https" {
+	if len(target.Scheme) == 0 || target.Scheme == "https" || target.Scheme == "http" {
```

Comment:
Quoting @smarterclayton from (see [original comment](https://github.com/openshift/origin/pull/15945#discussion_r137438176)):
> We don't want to send https credentials to http endpoints. You have to explicitly say you want to send credentials to an insecure endpoint.

If this is going to change, the comment below needs to be modified or removed.

Reply:
`credentialprovider.DockerKeyring` doesn't know anything about schemas. We are using it wrong, it works with image names, not with URLs.